pkg/controller/volume/persistentvolume/pv_controller.go

```diff
@@ -783,7 +784,20 @@ func (ctrl *PersistentVolumeController) updateClaimStatus(claim *v1.PersistentVo
 				return nil, fmt.Errorf("PersistentVolume %q is without a storage capacity", volume.Name)
 			}
 			claimCap, ok := claim.Status.Capacity[v1.ResourceStorage]
-			if !ok || volumeCap.Cmp(claimCap) != 0 {
+			// If PV has a resize annotation, set the claim's request capacity
+			if metav1.HasAnnotation(volume.ObjectMeta, util.AnnPreResizeCapacity) {
+				klog.V(2).Infof("volume %q requires filesystem resize: setting pvc %s status capacity to %s", volume.Name, claimToClaimKey(claim), volume.ObjectMeta.Annotations[util.AnnPreResizeCapacity])
+				preQty, err := resource.ParseQuantity(volume.ObjectMeta.Annotations[util.AnnPreResizeCapacity])
```

Comment:
I wonder if we can't parse here, we should simply revert to `volume.Spec.Capacity` rather than failing here.

Reply:
Sounds good.